arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/SelfdestructSection.java

```diff
@@ -69,27 +70,27 @@ public SelfdestructSection(Hub hub) {
     super(hub, (short) 8);
 
     // Init
-    this.id = hub.currentFrame().id();
-    this.transactionProcessingMetadata = hub.txStack().current();
-    this.hubStamp = hub.stamp();
-    this.exceptions = hub.pch().exceptions();
+    id = hub.currentFrame().id();
+    transactionProcessingMetadata = hub.txStack().current();
+    hubStamp = hub.stamp();
+    exceptions = hub.pch().exceptions();
 
     final MessageFrame frame = hub.messageFrame();
 
     // Account
-    this.address = frame.getSenderAddress();
-    this.accountBefore = AccountSnapshot.canonical(hub, this.address);
+    address = frame.getRecipientAddress();
+    selfdestructorAccountBefore = AccountSnapshot.canonical(hub, address);
 
     // Recipient
-    this.recipientRawAddress = frame.getStackItem(0);
-    this.recipientAddress = Address.extract((Bytes32) this.recipientRawAddress);
+    recipientRawAddress = frame.getStackItem(0);
+    recipientAddress = Address.extract((Bytes32) recipientRawAddress);
 
-    this.selfDestructTargetsItself = this.address.equals(this.recipientAddress);
+    selfDestructTargetsItself = address.equals(recipientAddress);
 
-    this.recipientAccountBefore =
+    recipientAccountBefore =
         selfDestructTargetsItself
-            ? this.accountAfter.deepCopy()
-            : AccountSnapshot.canonical(hub, this.recipientAddress);
+            ? selfdestructorAccountAfter.deepCopy()
+            : AccountSnapshot.canonical(hub, recipientAddress);
```

Comment:
**Question:** will this work for the deployment number and status ? We must have tests with SELFDESTRUCT's during CREATE(2)'s or during deployment transactions.